packages/compile-solidity/compilerSupplier/loadingStrategies/VersionRange.js

```diff
@@ -147,7 +195,7 @@ class VersionRange extends LoadingStrategy {
     events.emit("fetchSolcList:start", { attemptNumber: index + 1 });
```

Comment:
This whole function appears to be rolling its own retry logic.  It'd probably be better to tear that out and just use `axios-retry` like you're doing elsewhere.

Reply:
Opened [an issue](https://github.com/trufflesuite/truffle/issues/4261) for this